src/v/kafka/client/direct_consumer/tests/direct_consumer_fixture_test.cc

```diff
@@ -175,6 +175,29 @@ class consumer_fixture : public cluster_test_fixture {
           .get();
     }
 
+    // shuffle leadership, wait for leadership change to become visible to the
+    // test
+    void wait_for_visible_leadership_shuffle(const model::ntp& ntp) {
+        constexpr auto leadership_swap_wait = 300ms;
+        const auto original_leader = this->get_leader(ntp);
+        auto current_leader = original_leader;
+
+        this->shuffle_leadership(ntp).get();
+
+        for (const auto iteration : std::array{0, 1, 2, 3, 4, 5}) {
+            logger.info(
+              "polling for leadership change iteration: {}", iteration);
+            current_leader = this->get_leader(ntp);
+            ss::sleep(leadership_swap_wait).get();
+            if (current_leader != original_leader) {
+                break;
+            }
+        }
+        vassert(
+          current_leader != original_leader,
+          "never detected leadership change, test precondition failed");
```

Comment:
nitpick: since the this function returns void, you should be able to use gtest asserts here.